### PR TITLE
feat: add / edit schedule without calendar

### PR DESCRIPTION
**Pull request by tarnas14**
Description
---

- [x] form for editing schedules accessible by double click on the schedules list
- [x] allows to set values of the schedule except calendar 
- [ ] calendar (next pr #175 )
- [x] handles invalid time interval input
- [x] handles 'save' blocking until either weekdays or date is selected
- [x] use locales (ie. screen.getByText(t.mining.scheduling.add)) in scheduling tests instead of hardcoded text

TODO (next PRs)
- [ ] disabling scheduling link when neither mining is setup correctly
- [ ] disabling misconfigured mining types on schedule form

known issues:
- non-serializable data in the redux state (`Date`) - fixed in #175)

Motivation and Context
---

#43

How Has This Been Tested?
---

mining_scheduling_form

### CODE REVIEW

File: applications/launchpad/gui-react/src/components/Checkbox/Checkbox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    const onChange = jest.fn()
+    render(
+      <ThemeProvider theme={themes.light}>
+        <Checkbox checked={false} onChange={onChange}>
+          test
+        </Checkbox>
+      </ThemeProvider>,
+    )
+
+    const label = screen.getByText('test')
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/components/Checkbox/Checkbox.test.tsx:44]: NIT: It's good to define all identifiers at the beginning, ie:

```
const testText = 'test'

render(
      <ThemeProvider theme={themes.light}>
        <Checkbox checked={false} onChange={onChange}>
          {testText}
        </Checkbox>
      </ThemeProvider>,
    )

const label = screen.getByText(testText)
```

File: applications/launchpad/gui-react/src/locales/mining.ts [hunk 2]
```diff
@@ -28,6 +28,14 @@ const translations = {
       'Tari Launchpad must be open at the scheduled hours for mining to start.',
     noSchedules: 'No mining schedule has been set up yet',
     add: 'Add schedule',
+    removeSchedule: 'Remove schedule',
+    ops: 'Ops!',
+    error_miningEndsBeforeItStarts:
+      /* eslint-disable-next-line quotes */
+      "I guess you need to correct the hours because mining can't stop before it even starts",
+    error_miningEndsWhenItStarts:
+      /* eslint-disable-next-line quotes */
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/locales/mining.ts:37]: NIT: Maybe we can move this eslint rule at the beginning so it disables that for whole file.

**Review by tomaszantas — APPROVED** [review]
In overall, it looks good. Found one issue on mac that I describe below.

One issue on mac:

image

adding `max-width: 70px` on `.react-time-input-picker .input` solves the problem. Because I'm working on mac, I can fix this in another PR.

File: applications/launchpad/gui-react/src/locales/mining.ts (hunk 2 above)
Comment (tarnas14) [reply applications/launchpad/gui-react/src/locales/mining.ts:37]: I'd rather have that line by line, I prefer to disable linters as little as possible xD

File: applications/launchpad/gui-react/src/components/Checkbox/Checkbox.test.tsx (hunk 1 above)
Comment (tarnas14) [reply applications/launchpad/gui-react/src/components/Checkbox/Checkbox.test.tsx:44]: sure, if you prefer, will include in #175

**Pull request merged**